core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/variable/descriptor/ListVariableDescriptor.java

```diff
@@ -91,35 +76,36 @@ public Class<?> getElementType() {
     // ************************************************************************
 
     @Override
-    public boolean isInitialized(Object entity) {
-        return true;
-    }
-
-    public List<Object> getListVariable(Object entity) {
-        Object value = getValue(entity);
+    public List<Object> getValue(Object entity) {
+        Object value = super.getValue(entity);
         if (value == null) {
             throw new IllegalStateException("The planning list variable (" + this + ") of entity (" + entity + ") is null.");
         }
         return (List<Object>) value;
     }
 
     public Object removeElement(Object entity, int index) {
-        return getListVariable(entity).remove(index);
+        return getValue(entity).remove(index);
     }
 
     public void addElement(Object entity, int index, Object element) {
-        getListVariable(entity).add(index, element);
+        getValue(entity).add(index, element);
     }
 
     public Object getElement(Object entity, int index) {
-        return getListVariable(entity).get(index);
+        return getValue(entity).get(index);
     }
 
     public Object setElement(Object entity, int index, Object element) {
-        return getListVariable(entity).set(index, element);
+        return getValue(entity).set(index, element);
     }
 
     public int getListSize(Object entity) {
-        return getListVariable(entity).size();
+        return getValue(entity).size();
+    }
+
+    public ListVariableDataDemand<Solution_> getProvidedDemand() {
```

Comment:
This is a standard way of naming these things throughout the codebase, I did not invent it for this purpose.

What this means is that every list variable comes with a demand. Think of demand as a map key which retrieves a supply. (That's the reason why it's stored here; it needs to be unique and constant for every list var.) And the supply is the thing that holds all the information about the list variable's contents. (See `ExternalizedListVariableDataSupply`.)

Reply:
My issue is not with `Demand`, but with `Provided`, since it provides no details on what being demanded. It appears to be a merge of `SingletonInverseRelationDemand` and other related list property lookup. I'll be okay with `getListVariableDataDemand` since it describes what is being demanded.